### PR TITLE
Add collection management scripts for MOH TOSP

### PR DESCRIPTION
**Pull request by harishv7**
## Problem

Needed scripts to manage collection resources in the database for content migration.

Closes ISOM-1700

## Solution

**Breaking Changes**

- [x] No - this PR is backwards compatible

**Features**:

- Added script to backup collections and their resources to JSON files
- Added script to create new collections from local JSON files
- Added script to safely delete collections and their associated resources
- Added script to publish draft collections and their child resources

**New scripts**:

- `backupCollectionById.ts`: Backs up a collection and its resources to JSON files
- `createCollectionFromLocal.ts`: Creates a new collection from local JSON content
- `deleteCollectionById.ts`: Safely deletes a collection and all associated resources
- `publishDraftCollection.ts`: Publishes a draft collection and its child resources

## Tests

Scripts should be tested with:
1. Backup of an existing collection
2. Creation of a new collection from backed up files
3. Publishing the newly created collection
4. Deletion of test collections

Each script includes error handling and transaction support to maintain data integrity.

### CODE REVIEW

**Comment by harishv7** [comment]
* **#955** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/955?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/955?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: apps/studio/prisma/scripts/moh-tosp/publishDraftCollection.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { db } from "~/server/modules/database"
+import { FileLogger } from "../FileLogger"
+
+// Update the logger path if required
+const logger = new FileLogger("./publishCollectionById.log")
```
Comment (adriangohjw) [inline apps/studio/prisma/scripts/moh-tosp/publishDraftCollection.ts:5]: nit - should be `publishDraftCollection.log` for consistency across files

File: apps/studio/prisma/scripts/moh-tosp/createCollectionFromLocal.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+import { db, jsonb } from "~/server/modules/database"
+import { FileLogger } from "../FileLogger"
+
+// Update the logger path if required
+const logger = new FileLogger("./createCollectionFromLocal.log")
+
+export const createCollectionFromLocal = async (
+  contentDir: string,
+  siteId: number,
+  indexPageName: string, // should be placed outside the folder
```
Comment (adriangohjw) [inline apps/studio/prisma/scripts/moh-tosp/createCollectionFromLocal.ts:13]: not sure if i understand why this should be placed outside the folder?

**Review by adriangohjw — COMMENTED** [review]
Since this will only be used once a year:
- (nit) **Use Generated Enums**: Let's consider using generatedEnums (like ResourceType.Collection) instead of hardcoded strings. This will help keep the file updated with future changes, especially since it lacks tests.
- (nit) **Keyword Arguments**: Since many functions have multiple parameters, using keyword arguments over positional arguments might enhance clarity and safety.
- **Add a README**: Could you please add a simple README outlining the expected folder structure and usage steps? This will be helpful since this script is run infrequently.

Thank you!

File: apps/studio/prisma/scripts/moh-tosp/createCollectionFromLocal.ts (hunk 2 above)
Comment (harishv7) [reply apps/studio/prisma/scripts/moh-tosp/createCollectionFromLocal.ts:13]: This follows the same structure as what github based repos have due to filesystem based routing, the index page is outside the children folder

e.g. /hello-world.json -> index page
    /hello-world/ -> folder

**Comment by harishv7** [comment]
Agreed, had done a quick version earlier. Have added all the above!

**Review by adriangohjw — APPROVED** [review]
all clear, thank you!

**Pull request merged**